x-pack/plugins/security_solution/server/endpoint/services/actions/clients/sentinelone/sentinel_one_actions_client.ts

```diff
@@ -460,6 +467,95 @@ export class SentinelOneActionsClient extends ResponseActionsClientImpl {
     ).actionDetails;
   }
 
+  async getFileInfo(actionId: string, agentId: string): Promise<UploadedFileInfo> {
+    if (
+      !this.options.endpointService.experimentalFeatures.responseActionsSentinelOneGetFileEnabled
+    ) {
+      throw new ResponseActionsClientError(
+        `File downloads are not supported for ${this.agentType} agent type. Feature disabled`,
+        400
+      );
+    }
+    await this.ensureValidActionId(actionId);
+
+    const fileInfo: UploadedFileInfo = {
+      actionId,
+      agentId,
+      id: agentId,
+      agentType: this.agentType,
+      status: 'AWAITING_UPLOAD',
+      created: '',
+      name: '',
+      size: 0,
+      mimeType: '',
+    };
+
+    try {
+      const agentResponse = await this.fetchGetFileResponseEsDocForAgentId(actionId, agentId);
+
+      // Unfortunately, there is no way to determine if a file is still available in SentinelOne without actually
+      // calling the download API, which would return the following error:
+      // {  "errors":[ {
+      //      "code":4100010,
+      //      "detail":"The requested files do not exist. Fetched files are deleted after 3 days, or earlier if more than 30 files are fetched.",
+      //      "title":"Resource not found"
+      // } ] }
+      fileInfo.status = 'READY';
+      fileInfo.created = agentResponse.meta?.createdAt ?? '';
+      fileInfo.name = agentResponse.meta?.filename ?? '';
+      fileInfo.mimeType = 'application/octet-stream';
+    } catch (e) {
+      // Ignore "no response doc" error for the agent and just return the file info with the status of 'AWAITING_UPLOAD'
```

Comment:
Q: Does this mean that when there's an error the file status is stuck at `AWAITING_UPLOAD`?